experimental/SetPartitions/src/GenerateCategory.jl

```diff
@@ -0,0 +1,464 @@
+                # get fitting partitions in advance (improve runtime)
+                new_comp_temp_comp = Set()
+                if length(get_upper_points(i)) <= max_length
+                    new_comp_temp_comp = get(new_comp_by_size_top_bottom[2], length(get_upper_points(i)), -1)
+                end
+                if i in keys(without)
+                    operate_on = setdiff(new_comp_temp_comp, get(without, i, -1))
+                else
+                    operate_on = new_comp_temp_comp
+                end
+                for ii in operate_on
+                    if is_composable(i, ii) && is_worth_composition(i, ii, max_length)
+                        push!(to_comp, [i, ii])
+                        push!(already_c, [i, ii])
+                    end
+                    if is_composable(ii, i) && is_worth_composition(ii, i, max_length)
+                        push!(to_comp, [ii, i])
+                        push!(already_c, [ii, i])
+                    end
+                end
+            end
+        end
+
+        # do the compositions
+        al = copy(to_comp)
+
+        #println(to_comp)
+        
+        for (i, ii) in al
+            a = composition(i, ii)
+            if !(a in all_partitions)
+                trace[a] = ((i, ii), "c")
+                push!(all_partitions, a)
+
+                # call function which adds the partition a into the right set in the dicts
+                all_partitions_by_size = add_partition_to_dict(all_partitions_by_size, a)
+                all_partitions_by_size_top_bottom = add_partition_to_composition_dict(
+                    all_partitions_by_size_top_bottom, a)
+                new_comp_by_size_top_bottom = add_partition_to_composition_dict(new_comp_by_size_top_bottom, a)
+
+                stop_whole = false
+                push!(new_comp, a)
+                stop = false
+            else
+                # remove not fitting candidates for further iterations
+                pop!(to_comp, [i, ii])
+                if !(i in keys(without))
+                    without[i] = Set([ii])
+                else
+                    push!(get(without, i, -1), ii)
+                end
+            end
+        end
+    end
+    return [all_partitions, already_c, stop_whole, all_partitions_by_size, all_partitions_by_size_top_bottom, trace]
+end
+
+"""
+construct_category(p::Array, n::Int, tracing::Bool = false, max_artifical::Int = 0)
+
+This function outputs list of all partitions size n constructed from partitions in p (without using partitions of size
+greater than max(max(n, max(p)), max_artifical))
+
+# Arguments
+- `p`: list of partitions
+- `n`: size of partitions in constructing category
+- `tracing`: optinal input: activate tracing and get the output (category, trace)
+- `max_artifical`: optional input: allow partitions to grow greater while construction process
+
+# Returns
+- list of all partitions size n constructed from partitions in p
+
+# Examples
+```julia-repl
```

Comment:
Better us a `jldoctest`, then your example will automatically serve as a test for your code.